x-pack/test/saved_object_api_integration/common/suites/import.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from 'expect.js';
+import { SuperTest } from 'supertest';
+import { DEFAULT_SPACE_ID } from '../../../../plugins/spaces/common/constants';
+import { getIdPrefix, getUrlPrefix } from '../lib/space_test_utils';
+import { DescribeFn, TestDefinitionAuthentication } from '../lib/types';
+
+interface ImportTest {
+  statusCode: number;
+  response: (resp: { [key: string]: any }) => void;
+}
+
+interface ImportTests {
+  default: ImportTest;
+}
+
+interface ImportTestDefinition {
+  user?: TestDefinitionAuthentication;
+  spaceId?: string;
+  tests: ImportTests;
+}
+
+const createImportData = (spaceId: string) => [
+  {
+    type: 'dashboard',
+    id: `${getIdPrefix(spaceId)}a01b2f57-fcfd-4864-b735-09e28f0d815e`,
+    attributes: {
+      title: 'A great new dashboard',
+    },
+  },
+  {
+    type: 'globaltype',
+    id: '05976c65-1145-4858-bbf0-d225cc78a06e',
+    attributes: {
+      name: 'A new globaltype object',
+    },
+  },
+];
+
+export function importTestSuiteFactory(es: any, esArchiver: any, supertest: SuperTest<any>) {
+  const createExpectResults = (spaceId = DEFAULT_SPACE_ID) => async (resp: {
+    [key: string]: any;
+  }) => {
+    expect(resp.body).to.eql({
+      success: true,
+      successCount: 2,
+    });
+  };
+
+  const expectRbacForbidden = (resp: { [key: string]: any }) => {
+    expect(resp.body).to.eql({
+      statusCode: 403,
+      error: 'Forbidden',
+      message: `Unable to bulk_create dashboard,globaltype, missing action:saved_objects/dashboard/bulk_create,action:saved_objects/globaltype/bulk_create`,
+    });
+  };
+
+  const makeImportTest = (describeFn: DescribeFn) => (
+    description: string,
+    definition: ImportTestDefinition
+  ) => {
+    const { user = {}, spaceId = DEFAULT_SPACE_ID, tests } = definition;
+
+    describeFn(description, () => {
+      before(() => esArchiver.load('saved_objects/spaces'));
```

Comment:
Can we add a test case for when we try to import a file which we're only authorized to import some of the saved object types? We should be able to accomplish this by fabricating a saved object type, similar to what we do with the find APIs.

Reply:
I've added tests to cover this scenario, we can iterate on the approach when you get a chance.